analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -35,34 +35,53 @@ public override bool ShouldExecute()
         return walker.ContainsCondition;
     }
 
-    private sealed class SyntaxKindWalker : SafeVisualBasicSyntaxWalker
+    protected override bool IsInsideUsingDeclaration(SyntaxNode node) =>
+        node.IsKind(SyntaxKind.VariableDeclarator) && node.Parent.IsKind(SyntaxKind.UsingStatement);
+
+    protected override bool IsLockStatement(SyntaxNode syntax) =>
+        syntax.IsKind(SyntaxKind.SyncLockBlock);
+
+    internal sealed class SyntaxKindWalker : SafeVisualBasicSyntaxWalker
     {
         public bool ContainsCondition { get; private set; }
+
         public override void Visit(SyntaxNode node)
         {
             if (!ContainsCondition)
             {
-                ContainsCondition = node.IsAnyKind(
-                                    SyntaxKind.AndAlsoExpression,
-                                    SyntaxKind.BinaryConditionalExpression,
-                                    SyntaxKind.ConditionalAccessExpression,
-                                    SyntaxKind.DoWhileStatement,
-                                    SyntaxKind.DoUntilStatement,
-                                    SyntaxKind.IfStatement,
-                                    SyntaxKind.SelectStatement,
-                                    SyntaxKind.SimpleDoStatement,
-                                    SyntaxKind.TernaryConditionalExpression,
-                                    SyntaxKind.OrElseExpression,
-                                    SyntaxKind.WhileStatement);
-
                 base.Visit(node);
             }
         }
-    }
 
-    protected override bool IsInsideUsingDeclaration(SyntaxNode node) =>
-        node.IsKind(SyntaxKind.VariableDeclarator) && node.Parent.IsKind(SyntaxKind.UsingStatement);
+        public override void VisitBinaryConditionalExpression(BinaryConditionalExpressionSyntax node) =>
+            ContainsCondition = true;
 
-    protected override bool IsLockStatement(SyntaxNode syntax) =>
-        syntax.IsKind(SyntaxKind.SyncLockBlock);
+        public override void VisitBinaryExpression(BinaryExpressionSyntax node)
+        {
+            ContainsCondition |= node.Kind() is SyntaxKind.AndAlsoExpression or SyntaxKind.OrElseExpression;
+            base.VisitBinaryExpression(node);
+        }
+
```

Comment:
It seems like we miss `For` and `For Each` here

Reply:
see https://github.com/SonarSource/sonar-dotnet/pull/8215#discussion_r1381589899